src/mediaStreams.ts

```diff
@@ -0,0 +1,402 @@
+          break;
+          default:
+          break;
+        }
+      });
+
+      if (!reports.rttMS.hasOwnProperty('currentRoundTripTime')) {
+        if (!reports.rttMS.hasOwnProperty('roundTripTime')) {
+          reports.rttMS['currentRoundTripTime'] = self.preRTT['currentRoundTripTime'];
+        } else {
+          reports.rttMS['currentRoundTripTime'] = reports.rttMS['roundTripTime']; // for Firefox
+          delete reports.rttMS['roundTripTime'];
+        }
+      } else {
+        reports.rttMS['currentRoundTripTime'] = Math.round(reports.rttMS['currentRoundTripTime'] * 1000);
+      }
+
+      if (reports.rttMS.hasOwnProperty('currentRoundTripTime')) {
+        self.preRTT['currentRoundTripTime'] = reports.rttMS['currentRoundTripTime'];
+      }
+
+      if (self.session) {
+        if (self.session.onRTPStat) {
+          self.session.onRTPStat(reports, self.session);
+        } else if (self.onRTPStat) {
+          self.onRTPStat(reports, self.session);
+        } else {
+          self.session.emit('rtpStat', reports, self.session);
+        }
+      }
+
+    }).catch(error => {
+      this.rcWPLoge(self.ktag, JSON.stringify(error));
+    });
+  }
+
+  browser() {
+    if (navigator.userAgent.search('MSIE') >= 0) {
+      return  this.browsers['MSIE'];
+    } else if (navigator.userAgent.search("Chrome") >= 0) {
+      return this.browsers['Chrome'];
+    } else if (navigator.userAgent.search("Firefox") >= 0) {
+      return this.browsers['Firefox'];
+    } else if (navigator.userAgent.search("Safari") >= 0 && navigator.userAgent.search("Chrome") < 0) {
+      return this.browsers['Safari'];
+    } else if (navigator.userAgent.search("Opera") >= 0) {
+      return this.browsers['Opera'];
+    }
+    return 'unknown';
+  }
+
+  release() {
+    if (this.mediaStatsTimer) {
+      clearTimeout(this.mediaStatsTimer);
+      this.mediaStatsTimer = null;
+    }
+    if (this.session) {
+      this.session.sessionDescriptionHandler.removeListener('iceConnection', this.onStateChange);
+      this.session.sessionDescriptionHandler.removeListener('iceConnectionChecking', this.onStateChange);
+      this.session.sessionDescriptionHandler.removeListener('iceConnectionConnected', this.onStateChange);
+      this.session.sessionDescriptionHandler.removeListener('iceConnectionCompleted', this.onStateChange);
+      this.session.sessionDescriptionHandler.removeListener('iceConnectionFailed', this.onStateChange);
+      this.session.sessionDescriptionHandler.removeListener('iceConnectionDisconnected', this.onStateChange);
+      this.session.sessionDescriptionHandler.removeListener('iceConnectionClosed', this.onStateChange);
+    }
+  }
+
+  rcWPLoge(label, msg) {
+    if (this.session) {
+      this.session.logger.error(`${label} ${msg}`);
+    } else {
+      console.log(label, msg);
+    }
+  }
+
+  rcWPLogd(label, msg) {
+    if (this.session) {
+      this.session.logger.log(`${label} ${msg}`);
+    } else {
+      console.log(label, msg);
```

Comment:
This must be avoided. UA has logger: https://github.com/onsip/SIP.js/blob/master/src/UA.ts#L157